protocol/swap/status_manager.go

```diff
@@ -0,0 +1,63 @@
+package swap
+
+import (
+	"sync"
+
+	"github.com/athanorlabs/atomic-swap/common/types"
+)
+
+// statusManager provides lookup for the status channels. Status channels are
+// ephemeral between runs of swapd.
+type statusManager struct {
+	mu             sync.Mutex
+	statusChannels map[types.Hash]chan Status
+}
```

Comment:
I isolated the status channel management into this separate `statusManager` "class" and gave `swap.manager` an "is-a" relationship with it. The base swap manager was already using its mutex across multiple independent maps. Separating this, hopefully, lowers the code complexity.